packages/vite/src/node/plugins/workerImportMetaUrl.ts

```diff
@@ -144,18 +144,27 @@ export function workerImportMetaUrlPlugin(config: ResolvedConfig): Plugin {
             cleanString,
             index + allExp.length
           )
-          const file = path.resolve(path.dirname(id), rawUrl.slice(1, -1))
-          let url: string
+          const url = rawUrl.slice(1, -1)
+          workerResolver ??= config.createResolver({
+            tryIndex: false,
+            preferRelative: true
+          })
+          const file = (await workerResolver(url, id)) ?? url
```

Comment:
Would be great to apply the same `.` optimization here too.

Reply:
I can add the optimization but I just wanted to point out that I'll need to specify `extensions: []` in the resolver to keep consistent behavior in dev and build modes.

Reply:
Sounds good 👍